### ISSUE TITLE
[Project] URL based enhancements send invitations to contribute

### ISSUE THREAD

**Issue by znarf** [opencollective/opencollective#5631]
Related with https://github.com/opencollective/opencollective/issues/4893

Todo:
- [x] #5721
  - [x] URL should be updated with `amount` when it's changed. router.replace, not router.push 
  - [x] URL should be updated with `quantity` when it's changed.
  - [x] URL should be updated with `interval` when it's changed (month, year). Introduce an interval value for one-time.
  - [x] URL should be updated with `platformTip`? But maybe we don't want. Ignore it if not supported by the collective.
  - [x] URL should be updated with `contributeAs` when it's changed. It should be simply ignored if not supported for the user.
- [x] `quantity` should be supported
- [x] Check behavior of `interval` if not supported by a specific tier. Enforce API side.
- [x] Check behavior of `amount` if it's invalid on a specific tier and we pass a `contributeAs`
- [x] Rename `interval` to `frequency`?
- [x] `skipStepDetails` doesn't seem useful, linking to /profile is enough
- [x] Linking to /payment directly should be supported
- [x] People should be able to pass a default `email` and `name`.
- [x] Use shallow=true when replacing route (see https://nextjs.org/docs/routing/shallow-routing)
- [x] Ability to link directly to summary?

Will not be supported with this work:
- [ ] Invite an Organization (or non individual) to contribute to a Collective
- [ ] Draft Contribution? At payment step, decide to not pay now and receive a notification or a link
- [ ] Do something similar to Order confirmation (Strong Customer Authentication)
- [ ] Change or select a tier in the middle of the contribution flow
- [ ] Change or select a sub-account in the middle of the contribution flow
- [ ] Introduce an Order page?

**Comment by piamancini** [comment]
- [ ] Start implementation (@Betree)

**Comment by piamancini** [comment]
Status = Paused

**Comment by Betree** [comment]
I have started a PR in https://github.com/opencollective/opencollective-frontend/pull/7992 to update the URL with the state. I'm however facing an unanticipated constraint that increases the complexity of this issue: the state is currently decoded from the URL in `getInitialProps`. However, [shallow routing](https://nextjs.org/docs/routing/shallow-routing) (which we want to use to prevent unnecessary refetchs), skips those methods. 
We could implement quick hacks, but the proper solution to resolve this is to move the state decoding (e.g. in `contribution-flow/index`) which will have other benefits such as improved code sharing between embed and regular flow.

Another challenge that I faced is that we'll need to keep supporting the legacy routes, which adds some more complexity to this issue: https://github.com/opencollective/opencollective/issues/5716.

For these reasons, and because updating the URL doesn't seem like the most impactful thing to do for people who are requesting this feature, I'm going to focus on the other aspects and will eventually come back to the state thing later.
